### PR TITLE
Add port table to browse services page

### LINKED ISSUE #3662: Improve presentation of routing for services on overview page
The message is confusing. See  https://github.com/openshift/origin/issues/3610#issuecomment-119378462

### PR DESCRIPTION
**Pull request by spadgett**
Fixes #3662

Mobile:

openshift_web_console

Full:

openshift_web_console

Overview, removed wordy routing traffic message with not-so-useful IP.

openshift_web_console

### CODE REVIEW

**Comment by spadgett** [comment]
@jwforres ptal

**Comment by spadgett** [comment]
Related to #2378.

**Comment by spadgett** [comment]
There was a problem with my rebase. Please wait to review.

**Comment by spadgett** [comment]
Fixed. The `table-events` styles were being applied to all tables.

Also added the service type to browse services.

File: assets/app/styles/_tile.less [hunk 1]
```diff
@@ -14,10 +14,18 @@
   h1, h2, h3 {
     margin: (@line-height-computed / 2) 0px;
   }
-  .tile-details {
```
Comment (spadgett) [inline assets/app/styles/_tile.less]: `tile-details` was not used.

File: assets/app/styles/_core.less [hunk 2]
```diff
@@ -356,10 +356,10 @@ code {
   }
 }
 @media (max-width: @screen-xs-max) {
-  .table-events, thead, tbody, th, td, tr {
```
Comment (spadgett) [inline assets/app/styles/_core.less]: This style was being applied to all `thead`, `tbody`, ...

File: assets/app/scripts/filters/resources.js [hunk 3]
```diff
@@ -535,4 +535,14 @@ angular.module('openshiftConsole')
...
     };
+  })
+  .filter('serviceImplicitDNSName', function() {
+    return function(service) {
+      if (!service || !service.metadata || !service.metadata.name || !service.metadata.namespace) {
+        return '';
+      }
+
+      // cluster.local suffix is customizable, so leave it off. <name>.<namespace>.svc resolves.
+      return service.metadata.name + '.' + service.metadata.namespace + '.svc';
```
Comment (jwforres) [inline assets/app/scripts/filters/resources.js]: @liggitt what was the resolution on the issue with services using name.namespace, or did that only apply to the cluster.local DNS name

File: assets/app/views/project.html [hunk 4]
```diff
@@ -41,7 +41,7 @@ <h1 title="{{project | displayName}}">Project {{project | displayName}}</h1>
 
             <div class="component-block">
               <div class="component">
-                <div class="component-label">
+                <div ng-attr-title="{{service | serviceImplicitDNSName}}" class="component-label">
```
Comment (jwforres) [inline assets/app/views/project.html]: just curious why you are only using this for the title attribute?

File: assets/app/views/services.html [hunk 5]
```diff
@@ -35,6 +26,37 @@ <h3 id="service-{{service.metadata.name}}">{{service.metadata.name}}</h3>
...
+            </thead>
+            <tbody>
+              <tr ng-repeat="portMapping in service.spec.ports | orderBy:'port'">
+                <td>
+                  <span ng-if="portMapping.nodePort">{{portMapping.nodePort}}</span>
+                  <span ng-if="!portMapping.nodePort" class="text-muted">none</span>
+                </td>
+                <td role="presentation" class="text-muted">&#8594;</td>
+                <td><span ng-if="service.spec.portalIP && service.spec.portalIP !== 'None'">
+                    <span class="hidden-xs">{{service.spec.portalIP}}:</span></span>{{portMapping.port}}</td>
```
Comment (jwforres) [inline assets/app/views/services.html]: is this going to look redundant having the portalIP on every line with a lot of ports?

Comment (spadgett) [inline assets/app/views/services.html]: Yeah, I don't have a strong opinion. @liggitt suggested it originally. I can go either way.

openshift_web_console

File: assets/app/scripts/filters/resources.js (hunk 3 above)
Comment (liggitt) [inline assets/app/scripts/filters/resources.js]: It includes the namespace

File: assets/app/views/project.html (hunk 4 above)
Comment (spadgett) [inline assets/app/views/project.html]: My understand is that from within pods you can reference services by their short names like `database`, so the longer name isn't too important.

File: assets/app/views/services.html (hunk 5 above)
Comment (spadgett) [inline assets/app/views/services.html]: `portalIP` should be `clusterIP` in v1. I'll fix.

**Comment by jwforres** [comment]
[merge]

**Pull request merged**